src/App.jsx

```diff
@@ -91,32 +91,129 @@ import TodoList from "./TodoList"; // Import the new component
 
 // 7.21 Add a removeTodo handler function and pass it as a prop to TodoList.
 
-const App = () => {
-  const [todoList, setTodoList] = useState(() => {
-    const savedTodos = localStorage.getItem("savedTodoList");
-    return savedTodos ? JSON.parse(savedTodos) : [];
-  });
+// const App = () => {
+//   const [todoList, setTodoList] = useState(() => {
+//     const savedTodos = localStorage.getItem("savedTodoList");
+//     return savedTodos ? JSON.parse(savedTodos) : [];
+//   });
 
-  useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]);
+//   useEffect(() => {
+//     localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+//   }, [todoList]);
+
+//   const addTodo = (newTodo) => {
+//     setTodoList((prevTodos) => [...prevTodos, newTodo]);
+//   };
+
+//   const removeTodo = (id) => {
+//     const updatedList = todoList.filter((todo) => todo.id !== id);
+//     setTodoList(updatedList);
+//   };
 
+//   return (
+//     <>
+//       <h1>Todo List</h1>
+//       <AddTodoForm onAddTodo={addTodo} />
+//       <TodoList todoList={todoList} onRemoveTodo={removeTodo} />
+//     </>
+//   );
+// };
+
+// export default App;
+
+// 8.1 Remove Custom Hook
+
+export default function App() {
+  // 8.2 Remove
+  // // Use useState directly for managing todoList
+  // const [todoList, setTodoList] = useState(() => {
+  //   // Retrieve and parse the saved todo list from localStorage
+  //   const savedTodoList = localStorage.getItem("savedTodoList");
+  //   return savedTodoList ? JSON.parse(savedTodoList) : [];
+  // });
+  // 8.3 Set initial todoList state to an empty Array
+  const [todoList, setTodoList] = useState([]);
+  // 8.15 Added isLoading State
+  // // 8.6 State for Todo List and Loading
+  // const [loading, setLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(true);
+
+  // 8.4 Remove
+  // // Use useEffect to save the todoList to localStorage on changes
+  // useEffect(() => {
+  //   localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+  // }, [todoList]);
+
+  // 8.5 Simulate async data fetching
+  useEffect(() => {
+    // 8.7 change without reject
+    // const fetchData = new Promise((resolve, reject) => {
+    const fetchData = new Promise((resolve) => {
+      setTimeout(() => {
+        // Simulate initial todo list data
+        resolve({
+          data: {
+            todoList: [
```

Comment:
I think here we should be fetching the todoList data from the localStorage as the initial state of todoList was set earlier.
example:
JSON.parse(localStorage.getItem("savedTodoList")) || []